packages/signers/src/offchain-message-signer.ts

```diff
@@ -0,0 +1,72 @@
+import { OffchainMessageWithRequiredSignatories } from '@solana/offchain-messages';
+
+import { deduplicateSigners } from './deduplicate-signers';
+import { isMessageSigner, MessageSigner } from './message-signer';
+
+/**
+ * Represents a {@link Signer} that is required to sign an offchain message for it to be valid.
+ */
+export type OffchainMessageSignatorySigner<
+    TAddress extends string = string,
+    TSigner extends MessageSigner<TAddress> = MessageSigner<TAddress>,
+> = TSigner;
+
+/**
+ * An {@link OffchainMessage} type extension that allows {@link MessageSigner | MessageSigners} to
+ * be used as required signatories.
+ * *
+ * @typeParam TAddress - Supply a string literal to define a signatory having a particular address.
+ * @typeParam TSigner - Optionally provide a narrower type for {@link MessageSigner | MessageSigners}.
+ *
+ * @example
+ * ```ts
+ * import { OffchainMessage } from '@solana/offchain-messages';
+ * import { generateKeyPairSigner, InstructionWithSigners, OffchainMessageWithSigners } from '@solana/signers';
+ *
+ * const signer = await generateKeyPairSigner();
+ * const firstSignatory: OffchainMessageSignatory = { ... };
+ * const secondSignatory: OffchainMessageSignatorySigner = { ... };
+ * const offchainMessage: OffchainMessage & OffchainMessageWithSigners = {
+ *     /* ... *\/
+ *     requiredSignatories: [firstSignatory, secondSignatory],
+ * }
+ * ```
+ */
+export interface OffchainMessageWithSigners<
+    TAddress extends string = string,
+    TSigner extends MessageSigner<TAddress> = MessageSigner<TAddress>,
+> {
+    requiredSignatories: readonly OffchainMessageSignatorySigner<TAddress, TSigner>[];
```

Comment:
If I understand correctly, here you want to accept both signers and "address wrappers" as signatories. If so, what you had before makes more sense:

```suggestion
    requiredSignatories: readonly (OffchainMessageSignatory<TAddress> | OffchainMessageSignatorySigner<TAddress, TSigner>)[];
```